### PR TITLE
(6) fix(tracing): ReactNativeTracing and initial navigation spans have to be created after integrations setup

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## :loudspeaker: Type of change

- [x] Bugfix

## :scroll: Description

This PR removes the dependency of initializing tracing integrations like `NativeFramesTracking` and `StallTracking` before `ReactNativeTracing` integration.

This dependency was introduced in https://github.com/getsentry/sentry-react-native/pull/3996 and https://github.com/getsentry/sentry-react-native/pull/3997 when extracting the mentioned into standalone integration.

The cause of the dependency was that ReactNativeTracing start initiali navigation span during the integrations setup, but should do it after all integrations are setup.

## :green_heart: How did you test it?
integration tests, sample app 

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [x] All tests passing
- [x] No breaking changes

#skip-changelog

### CODE REVIEW

**Review by lucas-zimerman — APPROVED** [review]
LGTM!

File: src/js/tracing/reactnativetracing.ts [hunk 1]
```diff
@@ -111,6 +111,17 @@ export const reactNativeTracingIntegration = (
...
+
+    instrumentOutgoingRequests({
+      traceFetch: finalOptions.traceFetch,
+      traceXHR: finalOptions.traceXHR,
+      shouldCreateSpanForRequest: finalOptions.shouldCreateSpanForRequest,
+      tracePropagationTargets: client.getOptions().tracePropagationTargets || DEFAULT_TRACE_PROPAGATION_TARGETS,
+    });
+  };
+
+  const afterAllSetup = (): void => {
```
Comment (lucas-zimerman) [inline src/js/tracing/reactnativetracing.ts:124]: ● ReactNativeTracing › Tracing Instrumentation › With routing instrumentation › Cancels route transaction when app goes to background 

Seems to be failing on this specific PR, could you double-check if it broke with this changes?

Comment (lucas-zimerman) [inline src/js/tracing/reactnativetracing.ts:124]: ● ReactNativeTracing › Tracing Instrumentation › With routing instrumentation › Cancels route transaction when app goes to background 

Seems to be failing on this specific PR, could you double-check if it broke with this changes?

**Review by lucas-zimerman — CHANGES_REQUESTED** [review]

File: src/js/tracing/reactnativetracing.ts (hunk 1 above)
Comment (krystofwoldrich) [reply src/js/tracing/reactnativetracing.ts:124]: I forgot to update the test to call the newly added `afterAllSetup`. Thank you.

**Comment by krystofwoldrich** [comment]
I've fixed the JS tests.

The e2e fails due to https://github.com/getsentry/sentry-react-native/pull/4004 unrelated to this PR, it will be fixed when I merge main to v6.

**Review by lucas-zimerman — APPROVED** [review]
LGTM!

**Pull request merged**